packages/config-utils/README.md

```diff
@@ -1,198 +1,103 @@
-        root: resolve(__dirname, '../'),
-        // exposes: { './RootApp': './src/AppEntry.js' },
-        // shared: [{ react: { singleton: true, requiredVersion: 17.0.0 } }],
-        // debug: true,
-        // moduleName: 'someAppName',
-        // useFileHash: false,
-        // exclude: ['react']
-    })
-);
-```
+## 📚 Core Functions
 
-We've aimed for easy plug and play, but feel free to override any fed mods partials with these arguments:
+### Module Federation
+- **`federatedModules`** - Webpack module federation configuration for micro-frontends
+- **`generatePFSharedAssetsList`** - PatternFly dynamic modules discovery
 
-* `root` - root of you application (to find out `package.json`)
-* `exposes` - which files and modules should be exposed
-* `shared` - federated modules allow for sharing of dependencies, we've added some general dependencies, but feel free to expand them via this prop
-* `debug` - to see full output what is going to be used
-* `moduleName` - used to generate the file under this name (generally your `insights.appname` - if you have dashes in your name replace them with cammel case)
-* `useFileHash` - in order to prevent caching we are using file hashes, you might want to turn this off in your dev env
-* `exclude` - if you want to exclude any shared module you can do it trough here (even for general dependencies added by us)
-* `eager` - if set to `true`, turns eager fetch for shared modules on
+### Development Tools
+- **`proxy`** - Development server proxy for Red Hat environments
+- **`cookieTransform`** - Authentication cookie transformation
+- **`validateFrontendCrd`** - Frontend CRD validation
 
-### Override container ports
-The default ports for the RBAC server(**4012**) and/or the Keycloak server(**4001**) can be overridden using a corresponding variable in the environment. 
+### Utilities
+- **`searchIgnoredStyles`** - PatternFly CSS webpack aliases
+- **`jsVarName`** - String to JavaScript variable name conversion  
+- **`fecLogger`** - Logging utility with colored output
 
-  Bash examples:
+## 📖 Documentation
 
-      export KEYCLOAK_PORT=4020
-      export RBAC_PORT=5000
+Comprehensive documentation is available in the [`doc/`](./doc/) directory:
 
+- **[📋 Documentation Overview](./doc/README.md)** - Complete guide to all documentation files
+- **[⚙️ Core Functions](./doc/core-functions.md)** - Module federation, proxy, and validation
+- **[🛠️ Utility Functions](./doc/utility-functions.md)** - Helper utilities and tools
+- **[🚀 Advanced Features](./doc/advanced-features.md)** - FEO features and development workflows
 
-#### List of shared deps
+## 🏗️ Features
 
-This packages exposes these federated shared dependencies
-* `lodash` - version taken from your `package.json`
-* `axios` - version taken from your `package.json`
-* `redux` - version taken from your `package.json`
-* `react` - version taken from your `package.json`, treated as singleton
```

Comment:
I do feel like this format/wording made it a bit clearer to indicate what is in the shared scope - but you can in theory figure it out from the source. I'm fine with taking it out for now and seeing if anyone else misses it.